### PR TITLE
feature: Pill V3

### PR DESCRIPTION
**Pull request by rajsam003**
The type of this PR is: Feature

This PR resolves [[CX-1695](https://artsyproduct.atlassian.net/browse/CX-1695)]

### Description

This PR adds the new V3 Pill and uses them depending on the current theme.

imageimage

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add V3 Pill component (behind feature flag) - rajsam003

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Add V3 Pill component (behind feature flag) - rajsam003)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 398f5f56f5f66d731155306dee7adef64a37fd12
</p>

File: src/lib/Scenes/SavedSearchAlert/Components/Form.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ export const Form: React.FC<FormProps> = (props) => {
         <InputTitle>Filters</InputTitle>
         <Flex flexDirection="row" flexWrap="wrap" mt={1} mx={-0.5}>
           {pills.map((pill, index) => (
-            <Pill testID="alert-pill" m={0.5} key={`filter-label-${index}`}>
+            <Pill testID="alert-pill" size="xs" rounded m={0.5} key={`filter-label-${index}`}>
```
Comment (gkartalis) [inline src/lib/Scenes/SavedSearchAlert/Components/Form.tsx:69]: Could you remove `rounded` from this Pill? This is supposed to be square according to design. More info can be found on [this slack thread](https://artsy.slack.com/archives/C9SATFLUU/p1630572009268600)
```suggestion
            <Pill testID="alert-pill" size="xs" m={0.5} key={`filter-label-${index}`}>
```

**Review by gkartalis — CHANGES_REQUESTED** [review]

File: src/palette/elements/Pill/Pill.stories.tsx [hunk 2]
```diff
@@ -1,33 +1,43 @@
...
+      <Pill size="xs" icon={<CheckIcon fill="black100" mr={1} />}>
+        Checked
+      </Pill>
+      <Pill size="xs" icon={<CloseIcon fill="black100" ml={1} />} iconPosition="right">
+        Crossed
       </Pill>
     </List>
   ))
-  .add("rounded pill pressable inactive", () => (
+  .add("Text", () => (
```
Comment (gkartalis) [inline src/palette/elements/Pill/Pill.stories.tsx:30]: 🤩

File: src/palette/elements/Pill/Pill.tsx [hunk 3]
```diff
@@ -1,38 +1,70 @@
...
-          borderRadius,
-        },
-        style,
-      ]}
+      flexDirection="row"
+      justifyContent="center"
+      alignItems="center"
+      borderWidth={1}
+      borderRadius={icon || rounded ? 20 : 0}
+      borderColor={icon || selected ? color("black60") : color("black10")}
```
Comment (gkartalis) [inline src/palette/elements/Pill/Pill.tsx:55]: According to figma design the inactive color should be "black15"

File: src/palette/elements/Pill/Pill.tsx [hunk 4]
```diff
@@ -1,38 +1,70 @@
...
 }
 
-export const Pill: React.FC<PillProps> = ({ children, style, variant, active = true, onPress, ...other }) => {
-  const { colorV3, space } = useTheme()
-  const borderColor = active ? colorV3("black60") : colorV3("black15")
-  const borderRadius = variant === "textRound" ? 50 : 0
+export const Pill: React.FC<PillProps> = ({
+  children,
+  size = "xxs",
+  selected,
```
Comment (gkartalis) [inline src/palette/elements/Pill/Pill.tsx:37]: Maybe lets default this to `true` for now in order to have black60 as a borderColor in the case that nothing is passed as selected prop to ensure that it doesn't break the design of `src/lib/Scenes/SavedSearchAlert/Components/Form.tsx`

File: src/palette/elements/Pill/Pill.tsx [hunk 5]
```diff
@@ -1,38 +1,70 @@
...
+  rounded,
+  icon,
+  iconPosition = "left",
+  imageUrl,
+  onPress,
+  ...other
+}) => {
+  const { color, space } = useTheme()
```
Comment (gkartalis) [inline src/palette/elements/Pill/Pill.tsx:45]: this should be `colorV3` isn't it? 🤔
```suggestion
  const { colorV3, space } = useTheme()
```

**Review by olerichter00 — CHANGES_REQUESTED** [review]
That looks very good!

Comparing the pills in the app with the Figma design I think the padding on the right and left should be increased.

**Comment by gkartalis** [comment]
Agree with @olerichter00! after that lets merge it!

**Review by gkartalis — APPROVED** [review]
🚀

**Comment by gkartalis** [comment]
#squashOnGreen

**Review by olerichter00 — APPROVED** [review]
Thanks for the fix 🙇

**Pull request merged**